blocks/main_loop.py

```diff
@@ -42,6 +45,87 @@
 a model. Check documentation of your extensions."""
 
 
+class Timer(object):
```

Comment:
I agree the naming is awkward, I'll try and come up with something better, although I find `TimeAccounting` and `HierarchicalTimer` very obscure.